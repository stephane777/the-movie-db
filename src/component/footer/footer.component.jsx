import React from "react";
import sprite from "../../assets/img/sprite.svg";

import * as SF from "./footer.style";
const Footer = () => {
	return (
		<SF.FooterWrapper className="footer">
			<SF.Footer>
				<a
					href="https://github.com/stephane777"
					className="footer__link"
					target="_blank"
					rel="noreferrer"
				>
					<SF.Svg>
						<use
							href={`${sprite}#icon-github`}
							className="grid__icon-use"
						></use>
					</SF.Svg>
				</a>
				<SF.CreatedBy>Created by Stephane Candelas.</SF.CreatedBy>
			</SF.Footer>
		</SF.FooterWrapper>
	);
};
export { Footer };
