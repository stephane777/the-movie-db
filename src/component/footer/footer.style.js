import styled from "styled-components";

export const FooterWrapper = styled.div`
	height: 6rem;
	display: flex;
	padding-left: 4rem;
	background-color: var(--tmdbDarkBlue);
	display: flex;
	justify-content: center;
	align-items: center;
	margin-top: 12rem;
`;
export const Footer = styled.footer`
	display: flex;
	justify-content: center;
	align-items: center;
`;

export const CreatedBy = styled.p`
	color: lightgray;
	margin-left: 2rem;
	margin-bottom: 0;
	font-size: 1.2rem;
	font-weight: light;
`;

export const Svg = styled.svg`
	fill: lightgray;
	width: 2rem;
	height: 2rem;
`;
