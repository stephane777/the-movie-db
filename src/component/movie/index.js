export { Movie } from "./movie.component";
