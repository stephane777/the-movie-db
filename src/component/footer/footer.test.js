import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Footer } from "./footer.component";

describe("it should render the footer", () => {
	it("should render the footer", () => {
		const { container } = render(<Footer />);
		expect(container.firstChild.className).toMatch("footer");
	});
});

describe("SNAPSHOT", () => {
	it("should match the snapshot for App component", () => {
		const testedUnit = render(<Footer />);
		expect(testedUnit).toMatchSnapshot();
	});
});
