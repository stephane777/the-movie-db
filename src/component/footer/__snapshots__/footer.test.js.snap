// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT should match the snapshot for App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvaa jNKRLS footer"
      >
        <footer
          class="sc-gsDJrp hCHvzP"
        >
          <a
            class="footer__link"
            href="https://github.com/stephane777"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="sc-hKwCoD lbrBCT"
            >
              <use
                class="grid__icon-use"
                href="svg#icon-github"
              />
            </svg>
          </a>
          <p
            class="sc-dkPtyc kEOILh"
          >
            Created by Stephane Candelas.
          </p>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa jNKRLS footer"
    >
      <footer
        class="sc-gsDJrp hCHvzP"
      >
        <a
          class="footer__link"
          href="https://github.com/stephane777"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="sc-hKwCoD lbrBCT"
          >
            <use
              class="grid__icon-use"
              href="svg#icon-github"
            />
          </svg>
        </a>
        <p
          class="sc-dkPtyc kEOILh"
        >
          Created by Stephane Candelas.
        </p>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
